tools/deployment/k8s/s3/deploy.sh

```diff
@@ -0,0 +1,130 @@
+}
+
+wait_for_s3_availability() {
+    local max_retries=10
+    local retry_delay=6
+
+    local retries=0
+    while [ $retries -lt $max_retries ]; do
+        if ! aws s3 ls --endpoint-url "$AWS_ENDPOINT_URL" > /dev/null 2>&1; then
+            LOG "[INFO] S3 API endpoint did not return successfully. Retrying in $retry_delay seconds ..."
+            retries=$((retries+1))
+            sleep $retry_delay
+        else
+            break
+        fi
+    done
+
+    if [ $retries -eq $max_retries ]; then
+        LOG "[ERROR] Maximum retries reached. S3 API endpoint ${AWS_ENDPOINT_URL} didn't respond"
+        exit 1
+    fi
+}
+
+# Validate that mandatory environment variables are set
+for var in "AWS_ENDPOINT_URL" "LOG_VIEWER_BUCKET"; do
+    if [ -z "${!var}" ]; then
+        LOG "[ERROR] $var environment variable must be specified"
+        exit 1
+    fi
+done
+
+# Fetch RELEASE_TARBALL_URL via GitHub API and download release
+if [ -z "$TAG_NAME" ]; then
+    # If not defined, use latest prerelease
+    RELEASE_TARBALL_URL=$(curl --silent --show-error \
+      "https://api.github.com/repos/y-scope/yscope-log-viewer/releases" | \
+      jq -r 'map(select(.prerelease)) | first | .assets[0].browser_download_url')
+else
+    RELEASE_TARBALL_URL=$(curl --silent --show-error \
+      "https://api.github.com/repos/y-scope/yscope-log-viewer/releases/${TAG_NAME}" | \
+      jq -r '.assets[0].browser_download_url')
+fi
+if [ -z "$RELEASE_TARBALL_URL" ]; then
+  LOG "[ERROR] Could not automatically resolve release tarball URL"
+  exit 1
+fi
+
+# Wait until S3 endpoint is available
+LOG "[INFO] Waiting until s3://${LOG_VIEWER_BUCKET} endpoint becomes available ..."
+wait_for_s3_availability
+
+# Create log-viewer bucket if not already exist
+LOG "[INFO] Creating s3://${LOG_VIEWER_BUCKET} bucket"
+if ! aws s3api head-bucket --endpoint-url "$AWS_ENDPOINT_URL" --bucket "$LOG_VIEWER_BUCKET" 2>/dev/null; then
+    aws s3api create-bucket --endpoint-url "$AWS_ENDPOINT_URL" --bucket "$LOG_VIEWER_BUCKET"
+else
+    LOG "[WARN] Bucket s3://${LOG_VIEWER_BUCKET} already exists"
+fi
+
+# Define and apply the Bucket Policy for public read access
+LOG "[INFO] Applying public read access policy to s3://${LOG_VIEWER_BUCKET}"
+POLICY=$(cat <<EOP
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Principal": "*",
+            "Action": "s3:GetObject",
+            "Resource": "arn:aws:s3:::${LOG_VIEWER_BUCKET}/*"
+        }
+    ]
+}
+EOP
+)
+if ! aws s3api put-bucket-policy \
+  --endpoint-url "${AWS_ENDPOINT_URL}" --bucket "${LOG_VIEWER_BUCKET}" --policy "$POLICY"; then
+    LOG "[ERROR] Failed to set bucket policy for s3://${LOG_VIEWER_BUCKET}"
+    exit 1
+fi
+
+# Download and decompress release tarball to a temp directory
+LOG "[INFO] Downloading ${RELEASE_TARBALL_URL}"
+DECOMPRESSED_ASSETS_DIRECTORY="${DECOMPRESSED_ASSETS_DIRECTORY:-/tmp/yscope-log-viewer}"
```

Comment:
Instead of writing to `/tmp` which might conflict with existing code, can we use `mktemp`?